### ISSUE TITLE
Download assets

### ISSUE THREAD

**Issue by walshbr** [scholarslab/scholarslab.org#12]
Photos, podcasts, etc.

**Comment by walshbr** [comment]
Currently we have photos for the people, but the blog posts in particular have a lot of photos uploaded as media files that we need to download, store, and access.

**Comment by walshbr** [comment]
The download all assets button in wordpress is timing out in the interface. can we download these from the backend?

**Comment by mossiso** [comment]
easy, peasy. All of the media files are in the wp-contents/uploads/ directory, which are then further separated into year/month directories.

We need to think about where they need to go and in which folder structure. We could just use the same structure, http://scholarslab.org/[uploads|media|files|what-have-yous]/YYYY/MM/imagefile.jpg. Shouldn't be hard to find/replace file paths that need changing in existing files to what ever format we pick.

**Comment by amandavisconti** [comment]
@jeremyboggs @kbdonnally Does scholarslab.org/assets/img/posts/YYYY/MM/imagefile.jpg sound okay to you? @mossiso, wasn't sure if you were waiting on a structure decision...

**Comment by amandavisconti** [comment]
That sounds better to me. Would folks accidentally overwrite images by using the same file name? (If so, is there a rake task or other way to avoid that?)

http://scholarslab.org/media/imagefile.jpg

**Comment by jeremyboggs** [comment]
It's definitely possible.

I wonder if we should also look into some LFS approach, as suggested in #162. We've got a significant number of media files, not sure we should have those in the git repo.

**Comment by mossiso** [comment]
I can write a script to download the files from wp-content/uploads/ and modify the name with the year and month so as to avoid name collusion, and put them in a 'media' folder.

wp-content/uploads/2017/04/pretty-picture.jpg -> media/2017-04-pretty-picture.jpg
wp-content/uploads/2018/05/pretty-picture.jpg -> media/2018-05-pretty-picture.jpg

Also to keep in mind is that WordPress makes varying image sizes every time an image is uploaded. Do we want those or just the original?

Then, a script can easily go into each html file (in the Jekyll version) and change the referenced URL.

src="URL/wp-content/2017/04/pretty-picture.jpg" -> src="URL/media/2017-04-pretty-picture.jpg"

Other issues to think about: 
* workflow for people adding new media going forward
* naming convention for media going forward

**Comment by kbdonnally** [comment]
### Going forward:

I don't know anything about exporting stuff, but had some thoughts re. organizing future assets.

Naming convention could use the author name slug, to avoid overlap and allow people to easily re-use photos they've referenced before.* Would also allow location generation to make use of YAML front matter for whomever's writing the post.

### Example:
- Hand-written: `src="[url]/media/katherine-donnally/image-title.jpg"`
- Using frontmatter: `src="[url]/media/{{ author.slug }}/image-title.jpg"`
- If someone wants to put their images in front matter, they could also do that and have an easier time inserting them in the posts:

```yaml
images:
  - pic1: dogs.png
  - pic2: ducks.png
  - shortname: much-longer-image-title-if-needed.jpg
```
then, in my post:

```markdown
![my image](blah/media/{{ author.slug }}/{{ images.shortname}})

<!-- or -->
<img src='blah/media/{{ author.slug }}/{{ images.shortname}}' alt='my image'>
```
***
*I recognize that people won't just be referencing their own images, but it's a good start for findability.

**Comment by jeremyboggs** [comment]
1. Let's use static.scholarslab.org to house existing and future media assets;
2. Let's keep the `year/month/day` folder structure that WordPress used, for pre-migration content;
3. Let's use @kbdonnally suggested folder structure for content and media assets published going forward.

**Comment by amandavisconti** [comment]
@mossiso @jeremyboggs Would we be able to script moving these assets as described [here](https://github.com/scholarslab/scholarslab.org/issues/12#issuecomment-398839599) before 8/13? Let me know if not, and thanks!

**Comment by jeremyboggs** [comment]
Assets have been moved to static.scholarslab.org, and I think I've updated all refences to them in the codebase. If we find anything amiss, let's open a new ticket.

**Comment by walshbr** [comment]
@jeremyboggs just updated other tickets to reflect the need for this to be documented for people to author things themselves.
